index.js

```diff
@@ -0,0 +1,59 @@
+
+function calculate (input) {
+    //numberfy stringy numbers
+    for (let i=0; i < input.length; i++) {
+        const element = input[i];      
+        //check for strings
+        if ((typeof(element)) === 'string') {
+            //check for numbers in those strings
+            if (!isNaN(element)) {
+                // change stringy numbers to regular numbers
+                input[i] = parseInt(element);
+            }
+        }
+    }
+   
+    // removal of undefined and non-numerics 
+    // use filter() to return numbers and operators
+
+    function getNumsAndOperators(element){
+        if (element === '') {
+            return false;
+        }
+        return !isNaN(element) || element === '+' || element === '-' || element === '*' || element === '/'; 
+    }
+    input = input.filter(getNumsAndOperators);
```

Comment:
This isn't something that I'll take points off for, but if you wanted to practice writing/utilizing anonymous functions (fat arrow) then I'd recommend giving it a shot here, as lines 19-25 could be further condensed to just a line or two by doing so within the parameters of your filter().